diff --git a/node_modules/expo-notifications/android/build.gradle b/node_modules/expo-notifications/android/build.gradle
index 0be398b..b963c19 100644
--- a/node_modules/expo-notifications/android/build.gradle
+++ b/node_modules/expo-notifications/android/build.gradle
@@ -24,8 +24,9 @@ buildscript {
 
   // Ensures backward compatibility
   ext.getKotlinVersion = {
-    if (ext.has("kotlinVersion")) {
-      ext.kotlinVersion()
+    if (rootProject.ext.has("kotlinVersion")) {
+      def kv = rootProject.ext.get("kotlinVersion")
+      return (kv instanceof Closure) ? kv() : kv
     } else {
       ext.safeExtGet("kotlinVersion", "1.8.10")
     }
