diff --git a/node_modules/expo-notifications/android/build.gradle b/node_modules/expo-notifications/android/build.gradle
index 0be398b..a0b9b00 100644
--- a/node_modules/expo-notifications/android/build.gradle
+++ b/node_modules/expo-notifications/android/build.gradle
@@ -24,8 +24,9 @@ buildscript {
 
   // Ensures backward compatibility
   ext.getKotlinVersion = {
-    if (ext.has("kotlinVersion")) {
-      ext.kotlinVersion()
+    if (rootProject.ext.has("kotlinVersion")) {
+      def kv = rootProject.ext.get("kotlinVersion")
+      return (kv instanceof Closure) ? kv() : kv
     } else {
       ext.safeExtGet("kotlinVersion", "1.8.10")
     }
@@ -110,12 +111,13 @@ dependencies {
   }
 
   api 'androidx.core:core:1.5.0'
-  api 'androidx.lifecycle:lifecycle-runtime:2.2.0'
-  api 'androidx.lifecycle:lifecycle-process:2.2.0'
-  api 'androidx.lifecycle:lifecycle-common-java8:2.2.0'
+  api 'androidx.lifecycle:lifecycle-runtime:2.6.1'
+  api 'androidx.lifecycle:lifecycle-process:2.6.1'
+  api 'androidx.lifecycle:lifecycle-common:2.6.1'
 
 
-  api 'com.google.firebase:firebase-messaging:22.0.0'
+  // Commented out - version not available
+  // api 'com.google.firebase:firebase-messaging:23.2.1'
 
   api 'me.leolin:ShortcutBadger:1.1.22@aar'
 
